sdk/dotnet/Pulumi.Automation/LocalWorkspace.cs

```diff
@@ -34,13 +35,28 @@ public sealed class LocalWorkspace : Workspace
         private readonly LocalSerializer _serializer = new LocalSerializer();
         private readonly bool _ownsWorkingDir;
         private readonly Task _readyTask;
+        private static readonly SemVersion _minimumVersion = SemVersion.Parse("2.21.0");
 
         /// <inheritdoc/>
         public override string WorkDir { get; }
 
         /// <inheritdoc/>
         public override string? PulumiHome { get; }
 
+        private SemVersion? pulumiVersion;
+        /// <inheritdoc/>
+        public override SemVersion PulumiVersion 
+        { 
+            get 
+            { 
+                if (pulumiVersion == null)
+                {
+                    throw new ArgumentNullException("pulumiVersion");
```

Comment:
I would throw a different exception here - maybe `InvalidOperationException`.. the exceptions that are prefixed with `Argument...` are kind of reserved for like parameter validation at the beginning of a method